### PR TITLE
Checkbox Component now responds to baseFontSize prop

### PR DESCRIPTION
**Pull request by bruugey**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-1525](https://jira.mongodb.org/browse/LG-1525)
🎟 _Jira ticket:_ [LG-2625](https://jira.mongodb.org/browse/LG-2625)

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: packages/checkbox/src/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -22,36 +29,39 @@ import {
...
+import { CheckboxProps } from './Checkbox.types';
 
 /**
  * Checkboxes should be used whenever a user has an option they’d like to opt in or out of.
  *
  * Unlike toggles, checkboxes are used for actions, or features, that don’t immediately turn on or off. Checkboxes are usually found in forms as opposed to config pages.
  */
 function Checkbox({
-  darkMode: darkModeProp,
+  animate = true,
```
Comment (shaneeza) [inline packages/checkbox/src/Checkbox/Checkbox.tsx:40]: Do you wanna organize by defaults and aliases?

File: packages/checkbox/src/Checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -86,76 +96,81 @@ function Checkbox({
...
-        />
-
-        <Check
-          theme={theme}
-          isChecked={isChecked}
-          indeterminate={indeterminateProp}
+        <Label
+          id={labelId}
+          htmlFor={checkboxId}
+          darkMode={darkMode}
```
Comment (shaneeza) [inline packages/checkbox/src/Checkbox/Checkbox.tsx:116]: Do we still need `darkMode` here?

File: packages/checkbox/src/Checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -86,76 +96,81 @@ function Checkbox({
...
+            </span>
+          )}
+        </Label>
+
+        {description && (
+          <Description
+            className={cx(descriptionStyle, {
+              [disabledLabelDarkThemeOverrideStyle]: disabled && darkMode,
+            })}
+            darkMode={darkMode}
```
Comment (shaneeza) [inline packages/checkbox/src/Checkbox/Checkbox.tsx:166]: And here?

File: packages/checkbox/src/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (bruugey) [reply packages/checkbox/src/Checkbox/Checkbox.tsx:40]: i did it alphabetically but i can revert

File: packages/checkbox/src/Checkbox/Checkbox.tsx (hunk 2 above)
Comment (bruugey) [reply packages/checkbox/src/Checkbox/Checkbox.tsx]: good catch, rm'd

File: packages/checkbox/src/Checkbox/Checkbox.tsx (hunk 3 above)
Comment (bruugey) [reply packages/checkbox/src/Checkbox/Checkbox.tsx]: done!

File: packages/checkbox/src/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (shaneeza) [reply packages/checkbox/src/Checkbox/Checkbox.tsx:40]: I have no preference; whatever floats ur boat

**Review by shaneeza — APPROVED** [review]

**Pull request merged**